### ISSUE TITLE
Emojifying a country

### ISSUE THREAD

**Issue by zxcvbnmmohd** [petehouston/flutter-emoji#1]
I followed your example and it works but when I use it to emojify a country it just shows the full text instead of the emoji.

ex:

    EmojiParser().emojify('I :heart: :coffee:') // -> I ❤️ ☕
    EmojiParser().emojify('I :flag-ca: :coffee:')  // -> I :flag-ca: ☕

and when I print the getEmoji/Name and hasEmoji/Name functions on it both the emoji and the name respectively I get the following:

    I/flutter ( 7213): Emoji{name="flag-ca", full=":flag-ca:", code="🇨🇦"}
    I/flutter ( 7213): Emoji{name="flag-ca", full=":flag-ca:", code="🇨🇦"}
    true
    true

When I use the emoji directly it works. What am I doing wrong?

**Comment by zxcvbnmmohd** [comment]
Other regular emojis work though. Just the flags aren’t working.

**Comment by petehouston** [comment]
Strange though. Let me check.

**Comment by na2axl** [comment]
Hi @petehouston, the problem it's in your `REGEX_NAME` constant, it cannot detect names like `:flag-ca:` due to the underscore, the underscore is not a `\w` (word character) in regex.

I think changing the regex to `:[\w-]+:` may do the trick (I've not tested 😋)

**Comment by petehouston** [comment]
It is fixed in [2.2.0](https://github.com/petehouston/flutter-emoji/releases/tag/2.2.0). Thank you.

**Comment by trqalothman** [comment]
Hello @petehouston , Thank you for this great package, I have an issue and I do not really know why it is happening, here is what I am trying to achieve.  I want the user to select a flag, when the flag code is hard-coded in text widget, it is showing fine, however, when I try to use it inside a pop-up window some of the emojis disappear.  here is a screen shot:

[https://drive.google.com/file/d/10U5qi-C6KvCOKgwGrkJhSv62eH4eE_ra/view?usp=sharing](url)

in the screen shot, top right corner is ':flag-kw:' not working, however, on the screen that generates the popup it is displayed fine with the following code:

```
String _selectedFlag = ':flag-kw:';
GestureDetector(
                                   child: Text(
                                        '${parser.emojify(_selectedFlag)}',
                                        style: TextStyle(
                                          fontSize: 32,
                                        ),
                                      ),
                                      onTap: () {
                                        _showFlagsPicker(context);
                                      },
                                    )
```

and here is my code for the pop-up in case you would like to regenerate the issue:

```
List<String> _emojiCode = [
    ':flag-kw:',
    ':flag-sa:',
    ':flag-bh:',
    ':flag-qa:',
    ':flag-ae:',
    ':flag-om:',
    ':flag-af:',
    ':flag-al:',
    ':flag-dz:',
    ':flag-as:',
    ':flag-ad:',
    ':flag-ao:',
    ':flag-ai:',
    ':flag-aq:',
    ':flag-ag:',
    ':flag-ar:',
    ':flag-am:',
    ':flag-aw:',
    ':flag-au:',
    ':flag-at:',
    ':flag-az:',
    ':flag-bs:',
    ':flag-bd:',
    ':flag-bb:',
    ':flag-by:',
    ':flag-be:',
    ':flag-bz:',
    ':flag-bj:',
    ':flag-bm:',
    ':flag-bt:',
    ':flag-bo:',
    ':flag-bq:',
    ':flag-ba:',
    ':flag-bw:',
    ':flag-bv:',
    ':flag-br:',
    ':flag-io:',
    ':flag-bn:',
    ':flag-bg:',
    ':flag-bf:',
    ':flag-bi:',
    ':flag-cv:',
    ':flag-kh:',
    ':flag-cm:',
    ':flag-ca:',
    ':flag-ky:',
    ':flag-cf:',
    ':flag-td:',
    ':flag-cl:',
    ':flag-cn:',
    ':flag-cx:',
    ':flag-cc:',
    ':flag-co:',
    ':flag-km:',
    ':flag-cd:',
    ':flag-cg:',
    ':flag-ck:',
    ':flag-cr:',
    ':flag-hr:',
    ':flag-cu:',
    ':flag-cw:',
    ':flag-cy:',
    ':flag-cz:',
    ':flag-ci:',
    ':flag-dk:',
    ':flag-dj:',
    ':flag-dm:',
    ':flag-do:',
    ':flag-ec:',
    ':flag-eg:',
    ':flag-sv:',
    ':flag-gq:',
    ':flag-er:',
    ':flag-ee:',
    ':flag-sz:',
    ':flag-et:',
    ':flag-fk:',
    ':flag-fo:',
    ':flag-fj:',
    ':flag-fi:',
    ':flag-fr:',
    ':flag-gf:',
    ':flag-pf:',
    ':flag-tf:',
    ':flag-ga:',
    ':flag-gm:',
    ':flag-ge:',
    ':flag-de:',
    ':flag-gh:',
    ':flag-gi:',
    ':flag-gr:',
    ':flag-gl:',
    ':flag-gd:',
    ':flag-gp:',
    ':flag-gu:',
    ':flag-gt:',
    ':flag-gg:',
    ':flag-gn:',
    ':flag-gw:',
    ':flag-gy:',
    ':flag-ht:',
    ':flag-hm:',
    ':flag-va:',
    ':flag-hn:',
    ':flag-hk:',
    ':flag-hu:',
    ':flag-is:',
    ':flag-in:',
    ':flag-id:',
    ':flag-ir:',
    ':flag-iq:',
    ':flag-ie:',
    ':flag-im:',
    ':flag-il:',
    ':flag-it:',
    ':flag-jm:',
    ':flag-jp:',
    ':flag-je:',
    ':flag-jo:',
    ':flag-kz:',
    ':flag-ke:',
    ':flag-ki:',
    ':flag-kp:',
    ':flag-kr:',
    ':flag-kg:',
    ':flag-la:',
    ':flag-lv:',
    ':flag-lb:',
    ':flag-ls:',
    ':flag-lr:',
    ':flag-ly:',
    ':flag-li:',
    ':flag-lt:',
    ':flag-lu:',
    ':flag-mo:',
    ':flag-mg:',
    ':flag-mw:',
    ':flag-my:',
    ':flag-mv:',
    ':flag-ml:',
    ':flag-mt:',
    ':flag-mh:',
    ':flag-mq:',
    ':flag-mr:',
    ':flag-mu:',
    ':flag-yt:',
    ':flag-mx:',
    ':flag-fm:',
    ':flag-md:',
    ':flag-mc:',
    ':flag-mn:',
    ':flag-me:',
    ':flag-ms:',
    ':flag-ma:',
    ':flag-mz:',
    ':flag-mm:',
    ':flag-na:',
    ':flag-nr:',
    ':flag-np:',
    ':flag-nl:',
    ':flag-nc:',
    ':flag-nz:',
    ':flag-ni:',
    ':flag-ne:',
    ':flag-ng:',
    ':flag-nu:',
    ':flag-nf:',
    ':flag-mp:',
    ':flag-no:',
    ':flag-pk:',
    ':flag-pw:',
    ':flag-ps:',
    ':flag-pa:',
    ':flag-pg:',
    ':flag-py:',
    ':flag-pe:',
    ':flag-ph:',
    ':flag-pn:',
    ':flag-pl:',
    ':flag-pt:',
    ':flag-pr:',
    ':flag-mk:',
    ':flag-ro:',
    ':flag-ru:',
    ':flag-rw:',
    ':flag-re:',
    ':flag-bl:',
    ':flag-sh:',
    ':flag-kn:',
    ':flag-lc:',
    ':flag-mf:',
    ':flag-pm:',
    ':flag-vc:',
    ':flag-ws:',
    ':flag-sm:',
    ':flag-st:',
    ':flag-sn:',
    ':flag-rs:',
    ':flag-sc:',
    ':flag-sl:',
    ':flag-sg:',
    ':flag-sx:',
    ':flag-sk:',
    ':flag-si:',
    ':flag-sb:',
    ':flag-so:',
    ':flag-za:',
    ':flag-gs:',
    ':flag-ss:',
    ':flag-es:',
    ':flag-lk:',
    ':flag-sd:',
    ':flag-sr:',
    ':flag-sj:',
    ':flag-se:',
    ':flag-ch:',
    ':flag-sy:',
    ':flag-tw:',
    ':flag-tj:',
    ':flag-tz:',
    ':flag-th:',
    ':flag-tl:',
    ':flag-tg:',
    ':flag-tk:',
    ':flag-to:',
    ':flag-tt:',
    ':flag-tn:',
    ':flag-tr:',
    ':flag-tm:',
    ':flag-tc:',
    ':flag-tv:',
    ':flag-ug:',
    ':flag-ua:',
    ':flag-gb:',
    ':flag-um:',
    ':flag-us:',
    ':flag-uy:',
    ':flag-uz:',
    ':flag-vu:',
    ':flag-ve:',
    ':flag-vn:',
    ':flag-vg:',
    ':flag-vi:',
    ':flag-wf:',
    ':flag-eh:',
    ':flag-ye:',
    ':flag-zm:',
    ':flag-zw:',
    ':flag-ax:',
  ];

_showFlagsPicker(BuildContext ctx) {
    showDialog(
      // barrierColor: Colors.black54,
      context: ctx,
      builder: (_) => AlertDialog(
        backgroundColor: Theme.of(context).primaryColorDark,
        shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(9)),
        scrollable: true,
        content: Center(
          child: Container(
            color: Theme.of(context).primaryColorDark,
            height: MediaQuery.of(context).size.height / 2,
            width: MediaQuery.of(context).size.width * 0.96,
            padding: const EdgeInsets.all(0),
            child: GridView.count(
              padding: const EdgeInsets.all(0),
              crossAxisCount: 6,
              children: [
                for (var flag in _flags)
                  GestureDetector(
                    child: Text(
                      '${parser.emojify(flag)}',
                      style: TextStyle(
                        fontSize: 40,
                        // fontFamily: 'MarkaziText',
                      ),
                    ),
                    onTap: () {
                      setState(() {
                        _selectedFlag = flag;
                      });
                      Navigator.of(context).pop();
                    },
                  ),
              ],
            ),
          ),
          // ],
          // ),
        ),
      ),
    );
  }
```

Thank you :)
